### ISSUE TITLE
directory in containers

### ISSUE THREAD

**Issue by jebraieili** [apiato/apiato#470]
hi, i have three connection database in my project and i want separate these models and directories of each connections.
for example, the structure of the directories is as follows:
  &nbsp;app
  &nbsp;&nbsp;&nbsp;&nbsp;Containers
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Connection1
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;User
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Authentication
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Connection2
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Post
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Comment
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Connection3
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Tag
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;State

Or:

  &nbsp;app
  &nbsp;&nbsp;&nbsp;&nbsp;Connection1
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Containers
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;User
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Authentication
  &nbsp;&nbsp;&nbsp;&nbsp;Connection2
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Containers
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Post
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Comment
  &nbsp;&nbsp;&nbsp;&nbsp;Connection3
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Containers
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Tag
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;State

Is this possible? Or is there a better way to identify each connection separately?

**Comment by johannesschobel** [comment]
use proper format to explain the folder structure!

**Comment by deviouspk** [comment]
I think he means that he's using three different databases for his project.
Ask yourself is it necessary to use multiple database (are you combining
mongodb with mysql maybe)? I would just make multiple models. and clearly
name the model so it has a reference to the db connection (so you know what
db you are accessing).

Op ma 17 sep. 2018 om 12:24 schreef Johannes Schobel <
notifications@github.com>:

**Comment by johannesschobel** [comment]
you can force a `Model` to always use a specific `connection` by applying a 
``` php
protected $connection = 'your_custom_db_connect';
```

to your model, like described here ( https://laravel.com/docs/5.6/eloquent#eloquent-model-conventions )

**Comment by jebraieili** [comment]
I means that i using three different databases connection

**Comment by deviouspk** [comment]
well if you have the same data stored in multiple databases i would name your models with the name+connection behind it for example UserMysql & userMongo. Then in each modle you would specify protected $connection = 'mysql';
protected $connection = 'mongodb';

if you don't store the same data then it doesn't really matter right :)

**Comment by johannesschobel** [comment]
i would highly suggest NOT to store the same data / models in different databases.. this may become a dependency hell - and you have NO single-point-of-truth in your application. Which database is the "most recent one" - these are questions that may arise..

I would suggest to rethink this application scenario.. Maybe this is complete overkill what you are trying to do..

Please also consider that you may provide a dedicated application that handles all "graphQL Connections" - and let the "MySQL Connection" and "GraphQL Connection" communicate via REST.. may be more easy to use and maintain than the other approach.. not sure though

**Comment by deviouspk** [comment]
i completely agree with johannes here. Don't use multiple db's unless you have a very specific need to use multiple db connections (for example if you already have multiple databases running that other applications connect to.). It will be a hell to manage. Hard to maintain. Multiple points of failure. dependency hell. etc..
